src/cryptography/hazmat/backends/openssl/poly1305.py

```diff
@@ -0,0 +1,58 @@
+# This file is dual licensed under the terms of the Apache License, Version
+# 2.0, and the BSD License. See the LICENSE file in the root of this repository
+# for complete details.
+
+from __future__ import absolute_import, division, print_function
+
+
+from cryptography.exceptions import InvalidSignature
+from cryptography.hazmat.primitives import constant_time
+
+
+_POLY1305_TAG_SIZE = 16
+_POLY1305_KEY_SIZE = 32
+
+
+class _Poly1305Context(object):
+    def __init__(self, backend, key):
+        self._backend = backend
+
+        key_ptr = self._backend._ffi.from_buffer(key)
```

Comment:
Don't you need to keep this alive? I suspect you can get a UAF right now with `k = Poly1305(bytearray(range(16)); k.finalize()`

Reply:
Ugh, I refactored this for something else and didn't store the key itself. Yeah this can definitely UAF

Reply:
add a test please